### ISSUE TITLE
Re-implement Gravatar reset button without JavaScript

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#9972]
## Issue Summary

The avatar settings panel currently relies on JS so users can stop using their custom profile picture and instead revert to what they have set up on Gravatar:

https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/account/settings_panels/avatar.html

Reviewing this with @lb and @Lovelyfin00, we think we can make this work without JS, by putting a switch / checkbox component before the image, and using the `:checked` selector.

Sample code:

```html
<input type="checkbox" name="use-gravatar"> Revert to Gravatar
```

And sample CSS:

```css
[type="checkbox"][name="use-gravatar"]:checked ~ .avatar-panel__image .avatar {
    display: none;
}
```

**Comment by Lovelyfin00** [comment]
Hi @thibaudcolas @lb- 
So for the gravatar button issue, I don’t think it can be implemented with just scss
Because at the end of the day, the src of the img cannot be changed so even when I display the gravatar image by setting background-image, once I save the settings, it’s still the previous src that will show

**Comment by lb-** [comment]
@Lovelyfin00 - here is a basic POC HTML / Styles that could be used as a starting point.

Let's chat on a call about this in detaill

```html
{% load i18n wagtailadmin_tags %}
<style>
    .avatar-panel [type='checkbox']:checked ~ .avatar-panel__image {
        /* hide uploaded image if reset gravatar checked */
        border: 4px solid red;
    }

    .avatar-panel [type='checkbox']:not(:checked) ~ .gravatar-img {
        /* hide gravatar if reset gravatar is NOT checked */
        border: 4px solid blue;
    }
</style>
{% block content %}
    <div class="avatar-panel">
        {% if form.avatar.value %}
            <label class="w-field__label" for="avatar-avatar-clear_id" id="avatar-avatar-clear_id-label">{% trans "Reset to default" %}</label>
            <input type="checkbox" name="avatar-avatar-clear" id="avatar-avatar-clear_id">
            <img class="gravatar-img" src="{% avatar_url request.user 70 gravatar_only=True %}" alt=""/>
        {% endif %}
        <div class="avatar-panel__image">
            {% avatar user=request.user size="large" %}
        </div>
        <div class="avatar-panel__form">
            <label for="id_avatar-avatar">{% trans "Upload a profile picture:" %}</label>
            <input type="file" name="avatar-avatar" accept="image/*" id="id_avatar-avatar">
            <input type="hidden" name="avatar-avatar-clear" id="avatar-avatar-clear_id">
        </div>
    </div>
{% endblock %}
```

**Comment by lb-** [comment]
Refined solution - still needs some styling but should be closer.

As discussed the `on` value works 100% the same with the checkbox, so it's mostly a styling problem to get the column layout working.

https://developer.mozilla.org/en-US/docs/Web/CSS/:checked

Let's try to remove all / most of the styles in `client/scss/layouts/_account.scss` that relate to floats and instead use the flexbox tailwind classes.



```{% load i18n wagtailadmin_tags %}
<style>
    .avatar-panel .avatar-clear:checked ~ .avatar.uploaded {
        /* hide uploaded image if reset gravatar checked */
        display: none;
    }

    .avatar-panel .avatar-clear:not(:checked) ~ .avatar.gravatar {
        /* hide gravatar if reset gravatar is NOT checked */
        display: none;
    }
</style>
{% block content %}
    <div class="avatar-panel w-mt-5 w-flex w-flex-row w-flex-nowrap">
        <div class="w-something for width">
        {% if form.avatar.value %}
            <input class="avatar-clear" type="checkbox" name="avatar-avatar-clear" id="avatar-avatar-clear_id">
            <label class="w-field__label" for="avatar-avatar-clear_id" id="avatar-avatar-clear_id-label">{% trans "Reset to default" %}</label>
            <img class="avatar large gravatar" src="{% avatar_url request.user 70 gravatar_only=True %}" alt=""/>
        {% endif %}
        {% avatar user=request.user classname="uploaded" size="large" %}
        </div>
        <div class="w-field w-mt-5">
            <label for="id_avatar-avatar">{% trans "Upload a profile picture:" %}</label>
            <input type="file" name="avatar-avatar" accept="image/*" id="id_avatar-avatar">
        </div>
    </div>
{% endblock %}
```

<img width="974" alt="Screenshot 2023-02-01 at 9 39 56 pm" src="https://user-images.githubusercontent.com/1396140/216032873-3c47da8d-0f8a-4c7d-86e9-531078c6e97e.png">
<img width="936" alt="Screenshot 2023-02-01 at 9 39 59 pm" src="https://user-images.githubusercontent.com/1396140/216032887-a7657cb0-506f-4c89-8fc0-8f5e95e20c8f.png">
